mlflow/azureml/__init__.py

```diff
@@ -380,15 +382,49 @@ def deploy(
             _copy_file_or_tree(src=absolute_model_path, dst=model_directory_path),
         )
 
-        registered_model = AzureModel.register(
-            workspace=workspace, model_path=tmp_model_path, model_name=model_name, tags=tags
-        )
+        registered_model = None
+        azure_model_id = None
+
+        if model_uri.startswith("models:/"):
+            m_name = model_uri.split("/")[-2]
+            m_version = int(model_uri.split("/")[-1])
+            azure_model_id = "{}:{}".format(m_name, m_version)
+        elif (
+            model_uri.startswith("runs:/")
+            and get_tracking_uri().startswith("azureml")
+            and get_registry_uri().startswith("azureml")
+        ):
+            m = mlflow_register_model(model_uri, model_name)
+            azure_model_id = "{}:{}".format(m.name, m.version)
 
-        _logger.info(
-            "Registered an Azure Model with name: `%s` and version: `%s`",
-            registered_model.name,
-            registered_model.version,
-        )
+            _logger.info(
+                "Registered an Azure Model with name: `%s` and version: `%s`",
+                registered_model.name,
+                registered_model.version,
+            )
```

Comment:
I'm a bit confused by this logging statement. In this block, we're registering a model with the model registry, which I think is distinct from creating an `AzureModel`. Is this log message accurate?

Reply:
This is the inherent issue with representing a cloud object. Registering the model with the model registry does create the azure model, but it's a cloud object, not an in memory object.

When we register the model, it creates a cloud object in the azure model registry, and then because we're registering through MLFlow, that function returns an MLFlow SDK model object. However, we need an AzureML SDK model object to be able to deploy (at this time, that will change once we fully integrate with the deployment handling in MLFlow), so we end up getting the AzureML SDK object in order to do the deployment (the AzureML SDK uses the object constructor as a getter, it's a long story).